frontend/src/components/SearchByName.tsx

```diff
@@ -0,0 +1,79 @@
+import React, {useEffect, useState} from "react";
+import {getGameTitlesWithPrefix} from "../services/api.ts";
+import {Autocomplete, CircularProgress, TextField} from "@mui/material";
+import {GetGamesRequest} from "../data/GetGamesRequest.ts";
+
+interface SearchByNameProps {
+    platformFilter: string[];
+    onPrefixChange: (selectedPrefix: string) => void;
+}
+
+const SearchByName: React.FC<SearchByNameProps> = ({platformFilter, onPrefixChange}) => {
+    const [inputValue, setInputValue] = useState<string>('');
+    const [suggestedGames, setSuggestedGames] = useState<string[]>([]);
+    const [loading, setLoading] = useState(false);
+
+    useEffect(() => {
+        const fetchTitles = async () => {
+            if (inputValue.length < 2) {
+                setSuggestedGames([]);
+                return;
+            }
+
+            setLoading(true);
+
+            try {
+                const request: GetGamesRequest = {
+                    platforms: platformFilter,
+                    prefix: inputValue
+                }
+                const suggestions = await getGameTitlesWithPrefix(request);
```

Comment:
Du könntest dir hier mal [Tanstack Query](https://tanstack.com/query/latest) anschauen. Das ist eine Library, die dir viel von dem State Management bei API-Calls abnimmt. Wird in den meisten lise Projekten genutzt.

Reply:
Hab ich mich bisschen eingelesen, klingt spannend und wie etwas das ich auf jeden Fall gerne nutzen möchte. Würde aber in nem separaten PR passieren und dann ganzheitlich.